### ISSUE TITLE
Could you help me create a pre-commit hook?

### ISSUE THREAD

**Issue by MartinChavez** [denoland/deno_lint#238]
I am running the following, in the **pre-commit** file:

```bash
#!/bin/sh
FILES=$(git diff --cached --name-only --diff-filter=ACMR "*.ts" | sed 's| |\\ |g')
[ -z "$FILES" ] && exit 0

# Prettify all selected files
echo "$FILES" | xargs deno lint --unstable

# Add back the modified/prettified files to staging
echo "$FILES" | xargs git add

exit 0
```
It doesn't work, it stills commits the files, any ideas?

**Comment by MartinChavez** [comment]
Actually, what I am trying to do is to just fail the commit, not to fix anything.

**Comment by bartlomieju** [comment]
@MartinChavez you can check return code of `deno lint --unstable` - it will return code `1` if there's any diagnostic produced.

**Comment by MartinChavez** [comment]
@bartlomieju , tried that:

```bash
#!/bin/sh

STAGED_FILES=$(git diff --cached --name-only --diff-filter=ACMR "*.ts" | sed 's| |\\ |g')

if [[ "$STAGED_FILES" = "" ]]; then
  exit 0
fi

PASS=true

echo "\nValidating:\n"

for FILE in $STAGED_FILES
do
  deno lint --unstable "$FILE"

  if [[ "$?" == 1 ]]; then
    echo "\t\033[32mDenoLint Failed: $FILE\033[0m"
    PASS=false
  else
    echo "\t\033[41mDenoLint Passed: $FILE\033[0m"
  fi
done

echo "\nValidation completed!\n"

if ! $PASS; then
  echo "\033[41mCOMMIT FAILED:\033[0m Your commit contains files that should pass Deno Linter but do not. Please fix the errors and try again.\n"
  exit 1
else
  echo "\033[42mCOMMIT SUCCEEDED\033[0m\n"
fi

exit $?
```

It doesn't work, it succeds.

**Comment by bartlomieju** [comment]
@MartinChavez I'm not an expert in bash but I think it shouldn't be `if [[ "$?" == 1 ]]; then` but rather `if [[ $? == 1 ]]; then`. 

Also I'd recommend linting all files in one go, instead of linting file after file, it will be a lot faster.
